src/hydrusvideodeduplicator/dedup.py

```diff
@@ -79,22 +104,25 @@ def deduplicate(self, overwrite: bool = False, custom_query: list | None = None,
         self.hydlog.info("Deduplication done.")
 
     @staticmethod
-    def _calculate_perceptual_hash(video: str | bytes) -> str:
+    def _calculate_perceptual_hash(video: Path | str | bytes) -> str:
         perceptual_hash = Vpdq.vpdq_to_json(Vpdq.computeHash(video))
         assert perceptual_hash != "[]"
         return perceptual_hash
 
-    def _retrieve_video_hashes(self, search_tags) -> list:
+    def _retrieve_video_hashes(
+        self, search_tags: Iterable[str], file_service_keys: Iterable[str] | None = None
+    ) -> Iterable[str]:
         all_video_hashes = self.client.search_files(
             search_tags,
+            file_service_keys=file_service_keys,
```

Comment:
@appleappleapplenanner Out of curiosity, why'd you change this function in particular to take in the file_service_keys as an argument, instead of referencing `self.file_service_keys`? It works either way of course, just not sure why this has it as a function argument but `deduplicate()` references the class attribute.

Reply:
Just in case I want to use the function for other file services for whatever reason. I'm not even sure it's better to do it this way, but meh.